cors/handler.go

```diff
@@ -0,0 +1,94 @@
+package cors
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+type preflightInfo struct {
+	allowedMethods []string
+	allowedHeaders []string
+}
+
+type Middleware struct {
+	paths map[string]preflightInfo
+}
+
+func New() *Middleware {
+	return &Middleware{
+		paths: map[string]preflightInfo{},
+	}
+}
+
+func (m *Middleware) Middleware() func(http.Handler) http.Handler {
```

Comment:
Should this be a middleware? You inject some headers on all OPTIONS, but you still need to setup the route in Gorilla Mux? I know this was the case historically but one of the reasons I skipped it was that it felt like it should be handled somewhere else.

Have you considered just blanket implementing all headers for CORS? I think CORS is more for web servers and not APIs conceptually, but not sure.